library/common/extensions/key_value/platform/c_types.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "library/common/types/c_types.h"
+
+// NOLINT(namespace-envoy)
+
+#ifdef __cplusplus
+extern "C" { // function pointers
+#endif
+
+/**
+ * Function signature for reading value from implementation.
+ */
+typedef envoy_data (*envoy_kv_store_read_f)(envoy_data key, const void* context);
+
+/**
+ * Function signature for saving value to implementation.
+ */
+typedef void (*envoy_kv_store_save_f)(envoy_data key, envoy_data value, const void* context);
```

Comment:
Can this document what's expected as the `context` pointer?

Reply:
Sure.

Reply:
I added a more general description to the root c_types.h, since we use these everywhere.